sw/device/lib/crypto/drivers/entropy.c

```diff
@@ -0,0 +1,183 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/crypto/drivers/entropy.h"
+
+#include "sw/device/lib/base/abs_mmio.h"
+#include "sw/device/lib/base/bitfield.h"
+#include "sw/device/lib/base/memory.h"
+
+#include "csrng_regs.h"  // Generated
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+enum {
+  kBaseCsrng = TOP_EARLGREY_CSRNG_BASE_ADDR,
+
+  /**
+   * CSRNG genbits buffer size in uint32_t words.
+   */
+  kEntropyCsrngBitsBufferSize = 4,
```

Comment:
nit: it would be nice to see `NumWords` somewhere in the name.

Reply:
Done.